### PR TITLE
mpvScripts.smartskip: init at 0-unstable-2023-11-25

### PR DESCRIPTION
**Pull request by iynaix**
## Things done

Added [SmartSkip](https://github.com/Eisa01/mpv-scripts) that allows for skipping openings / endings using chapter markers or manually through silence detection.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/smartskip.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    repo = "mpv-scripts";
+    rev = "48d68283cea47ff8e904decc9003b3abc3e2123e";
+    sha256 = "sha256-95CAKjBRELX2f7oWSHFWJnI0mikAoxhfUphe9k51Qf4=";
+  };
+  passthru.scriptName = "SmartSkip.lua";
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = {
+    description = "Automatically or manually skip opening, intro, outro, and preview, like never before. Jump to next file, previous file, and save your chapter changes!";
```
Comment (FliegendeWurst) [inline pkgs/applications/video/mpv/scripts/smartskip.nix:24]:
```suggestion
    description = "Automatically or manually skip opening, intro, outro, and preview";
```

**Review by FliegendeWurst — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/applications/video/mpv/scripts/smartskip.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  version = "0-unstable-2023-11-25";
+
+  scriptPath = "scripts/SmartSkip.lua";
+  src = fetchFromGitHub {
+    owner = "Eisa01";
+    repo = "mpv-scripts";
+    rev = "48d68283cea47ff8e904decc9003b3abc3e2123e";
+    sha256 = "sha256-95CAKjBRELX2f7oWSHFWJnI0mikAoxhfUphe9k51Qf4=";
+  };
+  passthru.scriptName = "SmartSkip.lua";
```
Comment (uninsane) [inline pkgs/applications/video/mpv/scripts/smartskip.nix:19]:
```suggestion
```

when you specify `scriptPath`, then `buildLua` derives `scriptName` from the last component of that path, so specifying both here isn't necessary.

File: pkgs/applications/video/mpv/scripts/smartskip.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  lib,
+  fetchFromGitHub,
+  unstableGitUpdater,
+  buildLua,
+}:
+
+buildLua {
+  pname = "mpv-smartskip";
```
Comment (uninsane) [inline pkgs/applications/video/mpv/scripts/smartskip.nix:9]:
```suggestion
  pname = "smartskip";
```

File: pkgs/applications/video/mpv/scripts/smartskip.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  version = "0-unstable-2023-11-25";
+
+  scriptPath = "scripts/SmartSkip.lua";
+  src = fetchFromGitHub {
+    owner = "Eisa01";
+    repo = "mpv-scripts";
+    rev = "48d68283cea47ff8e904decc9003b3abc3e2123e";
+    sha256 = "sha256-95CAKjBRELX2f7oWSHFWJnI0mikAoxhfUphe9k51Qf4=";
```
Comment (uninsane) [inline pkgs/applications/video/mpv/scripts/smartskip.nix:17]:
```suggestion
    hash = "sha256-95CAKjBRELX2f7oWSHFWJnI0mikAoxhfUphe9k51Qf4=";
```

**Review by uninsane — CHANGES_REQUESTED** [review]
please fix the commit message to say `mpvScripts` instead of `mpvScrips` (without the 't')

File: pkgs/applications/video/mpv/scripts/smartskip.nix (hunk 4 above)
Comment (iynaix) [reply pkgs/applications/video/mpv/scripts/smartskip.nix]: Made the changes and fixed the commit message

**Review by uninsane — APPROVED** [review]

**Pull request merged**